internal/decryptor/decryptor.go

```diff
@@ -318,12 +318,10 @@ func (d *Decryptor) SetAuthOptions(ctx context.Context) {
 	case DecryptionProviderSOPS:
 		var opts []auth.Option
 
+		opts = append(opts, auth.WithClient(d.client))
+		opts = append(opts, auth.WithServiceAccountNamespace(d.kustomization.GetNamespace()))
 		if d.kustomization.Spec.Decryption.ServiceAccountName != "" {
-			serviceAccount := types.NamespacedName{
-				Name:      d.kustomization.Spec.Decryption.ServiceAccountName,
-				Namespace: d.kustomization.GetNamespace(),
-			}
-			opts = append(opts, auth.WithServiceAccount(serviceAccount, d.client))
+			opts = append(opts, auth.WithServiceAccountName(d.kustomization.Spec.Decryption.ServiceAccountName))
```

Comment:
In this case we need something like this:

```go
saName := d.kustomization.Spec.Decryption.ServiceAccountName
if saName == "" {
	saName = auth.GetDefaultDecryptionServiceAccount()
}
if saName != "" {
	opts = append(opts, auth.WithServiceAccountName(saName))
}
```

Reply:
Ouch! Good thing we added `saNamespace != ""` to `o.ShouldGetServiceAccountToken()`! We also need not to pass `auth.WithServiceAccountNamespace()` when `saName == ""`, otherwise we will allow `auth.defaultKubeConfigServiceAccout` to be used here!!

Reply:
Good catch!